### PR TITLE
set random seed

### PR DESCRIPTION
**Pull request by maxaalexeeva**

### CODE REVIEW

**Comment by maxaalexeeva** [comment]
@marcovzla should the seed come from configs? Also, does it make sense that when I run with different number of epochs in configs, I get different results? e.g., if I run multiple times with the number of epochs set to 1, I get the same results on epoch 1. If I set the num of epochs to 2, I get the same results on epoch 1 and epoch 2 between reruns, BUT the results on epoch 1 are different from those if I run with num of epochs=1.

**Review by marcovzla — CHANGES_REQUESTED** [review]
Yeah, I think the random seed should be declared in `config` and used here, i.e., `r_seed = config.SEED`
Otherwise this looks good!

**Review by marcovzla — APPROVED** [review]
Thanks!

**Comment by marcovzla** [comment]
I just tried it locally and it seems to be working.
Did you have issues with this?
also, the line `random.seed(r_seed)` is repeated, can you delete it before merging?
thanks!

**Comment by maxaalexeeva** [comment]
@marcovzla, it works with the same number of epochs, e.g., if i run with n_epochs=2, the results are always the same, but if I switch it to a different number of epochs, the results for epoch 1 and epoch 2 are not the same as when i run with n_epochs=2. I don't know if that's how it's supposed to be.

**Comment by marcovzla** [comment]
That may be because of the scheduler, which depends on the number of steps (num_batches * num_epochs)

See https://github.com/marcovzla/discobert/blob/07c5fa4a8bb47034ac95c2fd64028585867256b7/train.py#L36-L42

**Comment by maxaalexeeva** [comment]
@marcovzla, oh, I see. Makes sense. Thanks!
